src/server/search/search_family_test.cc

```diff
@@ -70,6 +70,73 @@ template <typename... Args> auto IsUnordArray(Args... args) {
   return RespArray(UnorderedElementsAre(std::forward<Args>(args)...));
 }
 
+MATCHER_P(IsMapMatcher, expected, "") {
+  if (arg.type != RespExpr::ARRAY) {
+    *result_listener << "Wrong response type: " << arg.type;
+    return false;
+  }
+
+  auto result = arg.GetVec();
+  if (result.size() != expected.size()) {
+    *result_listener << "Wrong resp array size: " << result.size();
+    return false;
+  }
+
+  using KeyValueArray = std::vector<std::pair<std::string, std::string>>;
+
+  KeyValueArray received_pairs;
+  for (size_t i = 0; i < result.size(); i += 2) {
+    received_pairs.emplace_back(result[i].GetString(), result[i + 1].GetString());
+  }
+
+  KeyValueArray expected_pairs;
+  for (size_t i = 0; i < expected.size(); i += 2) {
+    expected_pairs.emplace_back(expected[i], expected[i + 1]);
+  }
+
+  // Custom unordered comparison
+  std::sort(received_pairs.begin(), received_pairs.end());
+  std::sort(expected_pairs.begin(), expected_pairs.end());
+
+  return received_pairs == expected_pairs;
+}
+
+template <typename... Matchers> auto IsMap(Matchers... matchers) {
+  return IsMapMatcher(std::vector<std::string>{std::forward<Matchers>(matchers)...});
+}
+
+MATCHER_P(IsUnordArrayWithSizeMatcher, expected, "") {
```

Comment:
I think it is possible to do so with current utilities. Search for `RespArray(UnorderedElementsAre` in the codebase for examples.

Reply:
The problem is that the first int argument in the array always comes first, so it's ordered. The other elements of the array are unordered, that is what this matcher checks actually.

Reply:
i see